compiler/src/iree/compiler/Dialect/Stream/Transforms/test/unify_encoding_for_globals.mlir

```diff
@@ -681,3 +681,43 @@ module @same_encoding_twice_no_unification {
     util.return
   }
 }
+
+// -----
+
+// The constant is created via stream.tensor.constant with NamedParameterAttr
+// and directly encoded without being stored to a source global first.
+// Should unify based on the parameter name as the source key.
+
+#executable_target_vmvx_bytecode_fb = #hal.executable.target<"vmvx", "vmvx-bytecode-fb", {iree.encoding.resolver = #iree_encoding.specialization_resolver<123>}>
+#device_target_local = #hal.device.target<"local", {ordinal = 0 : index}, [#executable_target_vmvx_bytecode_fb]> : !hal.device
+#encoding1 = #iree_encoding.testing<layouts = [#iree_encoding.specialized<456>]>
+#encoding2 = #iree_encoding.testing<layouts = [#iree_encoding.specialized<789>]>
+
+//       CHECK:   util.global private @[[$DEVICE_A:.+]] =
+util.global private @device_a = #device_target_local
+util.global private @encoded_v1 : !stream.resource<constant>
+util.global private @encoded_v2 : !stream.resource<constant>
+
+// CHECK: util.initializer
+util.initializer {
+  // The constant is created with parameter attribute, not loaded from a global.
+  // CHECK: %[[CST:.+]] = stream.tensor.constant
```

Comment:
Should we check here that the constant still uses the same name?

Reply:
The answer is no to me. It is clear that the pass does not update parameters.

And here is the guide: https://mlir.llvm.org/getting_started/TestingGuide/#filecheck-best-practices

> Tests should be minimal, and only check what is absolutely necessary.